source/extensions/common/wasm/wasm.cc

```diff
@@ -78,7 +78,9 @@ Wasm::Wasm(WasmConfig& config, absl::string_view vm_key, const Stats::ScopeShare
           createWasmVm(config.config().vm_config().runtime()), config.config().vm_config().vm_id(),
           MessageUtil::anyToBytes(config.config().vm_config().configuration()),
           toStdStringView(vm_key), config.environmentVariables(), config.allowedCapabilities()),
-      scope_(scope), cluster_manager_(cluster_manager), dispatcher_(dispatcher),
+      scope_(scope), stat_name_set_(scope_->symbolTable().makeSet("wasm")),
```

Comment:
Use a StatNamePool rather than a StatNameSet. StatNameSet has extra map overhead so you can translate "wasm" into its symbol after it was created, but you're not using that.